doc/guidelines/frontend-api-calls.md

```diff
@@ -0,0 +1,68 @@
+# Frontend API Calls
+
+These guidelines outline best practices for making API calls using React-Query within our project.
+
+## What is React-Query?
+
+[React-Query](https://react-query.tanstack.com/) is a library that provides hooks and utilities for managing, caching, and synchronizing asynchronous and remote data in React applications.
+
+## Basic Usage
+
+When integrating React-Query for API calls, consider the following guidelines:
+
+### Query keys
+
+- Use meaningful and unique query keys to identify each API call. The query key is crucial for React-Query to manage caching and data fetching efficiently.
+- Have a look at following guidelines: [https://react-query.tanstack.com/docs/guides/query-keys](https://react-query.tanstack.com/docs/guides/query-keys)
+
+#### Example
+
+```javascript
+const useUserData = (userId) => {
+  return useQuery(["user", userId], fetchUserData);
+};
+```
+
+### Error handling
```

Comment:
Consider adding that handling the error locally e.g. for form fields makes sense and notifications should be a last resort.

Reply:
i seperated local error handling and global callbacks in the guideline